plugins/bulk-import-backend/src/schema/openapi.yaml

```diff
@@ -210,37 +168,53 @@ paths:
         - BearerAuth: []
       tags: [Import]
       parameters:
-        - in: query
-          name: pagePerIntegration
-          description: the page number for each Integration
-          schema:
-            type: integer
-            default: 1
-        - in: query
-          name: sizePerIntegration
-          description: the number of items per Integration to return per page
-          schema:
-            type: integer
-            default: 20
-        - in: query
-          name: search
-          description: returns only Imports that contain the search string, by repository name
-          schema:
-            type: string
+        - $ref: '#/components/parameters/apiVersionHeaderParam'
+
+        # The '*PerIntegration' query params are being kept for backward compatibility,
+        # but the behavior depends on the API Version specified in the headers.
+        # As of API version v2, 'page' and 'size' parameters will always take precedence
+        # over the `*PerIntegration' params.
+        - $ref: '#/components/parameters/pagePerIntegrationQueryParam'
+          # TODO(rm3l): Overriding properties does not seem to work
```

Comment:
It works for me:
![image](https://github.com/user-attachments/assets/a6154e36-547b-4565-a314-7a47f9913bb2)

Added deprecated directly to the parameter, e.g:
```
    pagePerIntegrationQueryParam:
      in: query
      name: pagePerIntegration
      description: the page number for each Integration
      schema:
        type: integer
        default: 1
      deprecated: true
```

Reply:
Yes, it works when put in the `components.parameters` section, but then it would apply to all the other paths, even where they should not be deprecated.
Conversely, I tried overriding it for the `findAllImports` path only, like so:
```diff
diff --git a/plugins/bulk-import-backend/src/schema/openapi.yaml b/plugins/bulk-import-backend/src/schema/openapi.yaml
index 448288e9..eed87bab 100644
--- a/plugins/bulk-import-backend/src/schema/openapi.yaml
+++ b/plugins/bulk-import-backend/src/schema/openapi.yaml
@@ -176,11 +176,11 @@ paths:
         # over the `*PerIntegration' params.
         - $ref: '#/components/parameters/pagePerIntegrationQueryParam'
           # TODO(rm3l): Overriding properties does not seem to work
-        #          deprecated: true
+          deprecated: true
         #          description: Deprecated. Set the 'API-Version' request header to 'v2' and use the 'page' query parameter instead.
         - $ref: '#/components/parameters/sizePerIntegrationQueryParam'
           # TODO(rm3l): Overriding properties does not seem to work
-        #          deprecated: true
+          deprecated: true
         #          description: Deprecated. Set the 'API-Version' request header to 'v2' and use the 'size' query parameter instead.
         - $ref: '#/components/parameters/pageQueryParam'
         - $ref: '#/components/parameters/sizeQueryParam'
~
```

It rendered correctly in the OpenAPI UI, but for some reason, the backend refuses to start:

<details>
<summary>Logs</summary>

```shell
# Regenerate OpenAPI
$ yarn workspace @janus-idp/backstage-plugin-bulk-import-backend openapi
[...]
$ yarn workspace @janus-idp/backstage-plugin-bulk-import-backend start 

yarn workspace v1.22.20                                                                                                                                                                        
yarn run v1.22.20                                                                                                                                                                              
[...]
2024-10-04T13:38:13.444Z catalog info Performing database migration                                                                                                                            
Error: Document is not valid OpenAPI. 10 validation errors: 
[                                                                                                                                                                                              
  {                                                                                                                                                                                            
    "instancePath": "/paths/~1imports/get/parameters/1",                                                                                                                                       
    "schemaPath": "#/required",                                                                                                                                                                
    "keyword": "required",                                                                                                                                                                     
    "params": {                                                                                                                                                                                
      "missingProperty": "name"                                                                                                                                                                
    },                                                                                                                                                                                         
    "message": "must have required property 'name'"                                                                                                                                            
  },                                                                                                                                                                                           
  {                                                                                                                                                                                            
    "instancePath": "/paths/~1imports/get/parameters/1",                                                                                                                                       
    "schemaPath": "#/required",                                                                                                                                                                
    "keyword": "required",                                                                                                                                                                     
    "params": {                                                                                                                                                                                
      "missingProperty": "in"                                                                                                                                                                  
    },                                                                                                                                                                                         
    "message": "must have required property 'in'"                                                                                                                                              
  },                                                                                                                                                                                           
  {                                                                                                                                                                                            
    "instancePath": "/paths/~1imports/get/parameters/1",
    "schemaPath": "#/additionalProperties",
    "keyword": "additionalProperties",
    "params": {
      "additionalProperty": "$ref"
    },
    "message": "must NOT have additional properties"
  },
  {
    "instancePath": "/paths/~1imports/get/parameters/1",
    "schemaPath": "#/additionalProperties",
    "keyword": "additionalProperties",
    "params": {
      "additionalProperty": "deprecated"
    },
    "message": "must NOT have additional properties"
  },
  {
    "instancePath": "/paths/~1imports/get/parameters/1",
    "schemaPath": "#/oneOf",
    "keyword": "oneOf",
    "params": {
      "passingSchemas": null
    },
    "message": "must match exactly one schema in oneOf"
  },
  {
    "instancePath": "/paths/~1imports/get/parameters/2",
    "schemaPath": "#/required",
    "keyword": "required",
    "params": {
      "missingProperty": "name"
    },
    "message": "must have required property 'name'"
  },
  {
    "instancePath": "/paths/~1imports/get/parameters/2",
    "schemaPath": "#/required",
    "keyword": "required",
    "params": {
      "missingProperty": "in"
    },
    "message": "must have required property 'in'"
  },
  {
    "instancePath": "/paths/~1imports/get/parameters/2",
    "schemaPath": "#/additionalProperties",
    "keyword": "additionalProperties",
    "params": {
      "additionalProperty": "$ref"
    },
    "message": "must NOT have additional properties"
  },
  {                                                                                                                                                                         15:38:13 [168/3864]
    "instancePath": "/paths/~1imports/get/parameters/2",
    "schemaPath": "#/additionalProperties",
    "keyword": "additionalProperties",
    "params": {
      "additionalProperty": "deprecated"
    },
    "message": "must NOT have additional properties"
  },
  {
    "instancePath": "/paths/~1imports/get/parameters/2",
    "schemaPath": "#/oneOf",
    "keyword": "oneOf",
    "params": {
      "passingSchemas": null
    },
    "message": "must match exactly one schema in oneOf"
  }
]
    at OpenAPIBackend.validateDefinition (/home/asoro/work/projects/backstage/janus-idp/backstage-plugins/node_modules/openapi-backend/src/backend.ts:652:13)
    at OpenAPIBackend.init (/home/asoro/work/projects/backstage/janus-idp/backstage-plugins/node_modules/openapi-backend/src/backend.ts:209:14)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at createRouter (/home/asoro/work/projects/backstage/janus-idp/backstage-plugins/plugins/bulk-import-backend/src/service/router.ts:127:3)
    at Object.init (/home/asoro/work/projects/backstage/janus-idp/backstage-plugins/plugins/bulk-import-backend/src/plugin.ts:54:24)
    at <anonymous> (/home/asoro/work/projects/backstage/janus-idp/backstage-plugins/node_modules/@backstage/backend-app-api/src/wiring/BackendInitializer.ts:290:11)
    at async Promise.all (index 1)
    at BackendInitializer.#doStart (/home/asoro/work/projects/backstage/janus-idp/backstage-plugins/node_modules/@backstage/backend-app-api/src/wiring/BackendInitializer.ts:236:5)
    at BackendInitializer.start (/home/asoro/work/projects/backstage/janus-idp/backstage-plugins/node_modules/@backstage/backend-app-api/src/wiring/BackendInitializer.ts:150:5)
    at BackstageBackend.start (/home/asoro/work/projects/backstage/janus-idp/backstage-plugins/node_modules/@backstage/backend-app-api/src/wiring/BackstageBackend.ts:42:11)
Unknown operationId ping
Unknown operationId findAllOrganizations
Unknown operationId findAllRepositories
Unknown operationId findRepositoriesByOrganization
Unknown operationId findAllImports
Unknown operationId createImportJobs
Unknown operationId findImportStatusByRepo
Unknown operationId deleteImportByRepo

```

</details>

Reply:
`pagePerIntegration` and `sizePerIntegration` are deprecated only on the `GET /imports` v2 API, but still make sense atm on the other endpoints where pagination depends on the GH integrations configured.

Reply:
With hindsight, I thought that it might not make sense to set `deprecated: true` because it actually depends on the value of the `api-version` header, which is only known when handling a user request. Deprecated for v2, but not v1. So we might not be able to easily convey that in the OpenAPI spec.

Reply:
Oh, I totally missed the other paths!
I thought that deprecated for `v1` signals that it won't be supported in future versions? Yeah, that might be hard to define in the schema. Maybe it could be done via using copy of `pagePerIntegrationQueryParam` and `sizePerIntegrationQueryParam` just for this endpoint that will have `deprecated: True`. Although it won't look nice.

Reply:
> I thought that deprecated for v1 signals that it won't be supported in future versions?

Yes, that's the goal, but not sure when that will happen. But it will need to be maintained anyway.
Okay, let's copy the query parameter and mark it as deprecated. At least, it makes things clearer when looking at the parameter on the specific `findAllImports` path.
I'll update it soon. Thanks.

Reply:
Done at [`773b119` (#2246)](https://github.com/janus-idp/backstage-plugins/pull/2246/commits/773b1199566435fb8994411ff0c877c569e19309). PTAL. Thanks.